freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -77,6 +79,28 @@
         }, LOGGER, ":::Auto sync::: provider is checked in {}ms");
     }
 
+    public Set<InstanceMetaData> updateStatuses(Stack stack, InstanceStatus instanceStatus, boolean updateStatusFromFlow) {
+        Set<InstanceMetaData> checkableInstances = stack.getAllInstanceMetaDataList().stream()
+                .filter(Predicate.not(InstanceMetaData::isTerminated))
+                .filter(Predicate.not(InstanceMetaData::isDeletedOnProvider))
+                .collect(Collectors.toSet());
+
+        if (!updateStatusFromFlow && flowLogService.isOtherFlowRunning(stack.getId())) {
+            throw new InterruptSyncingException("interrupt updating instance status in updateStatuses, flow is running on freeipa stack " +
+                    stack.getName());
+        } else {
+            checkableInstances.forEach(instanceMetaData -> {
+                if (updateStatus) {
```

Comment:
as this is not part of the sync, it should be always updated when the instances are stopped. So this condition is not necessary

Reply:
Yes. You are right. I removed it.